src/arena.rs

```diff
@@ -60,11 +58,16 @@ impl<T: Sync + Send, const BITARRAY_LEN: usize, const LEN: usize> Default
 const fn check_const_generics<const BITARRAY_LEN: usize, const LEN: usize>() {
     let bits = usize::BITS as usize;
 
-    cfn_assert!(LEN <= (u32::MAX as usize));
-    cfn_assert_eq!(LEN % bits, 0);
-    cfn_assert_ne!(LEN, 0);
-
-    cfn_assert_eq!(LEN / bits, BITARRAY_LEN);
+    assert!(
+        LEN <= (u32::MAX as usize),
+        "LEN must be less than or equal to u32::MAX"
+    );
```

Comment:
We can avoid the unchecked `as usize` with the following code:

```suggestion
    let msg = "LEN must be less than or equal to u32::MAX";
    let len: u32 = LEN.try_into().expect(msg);
    assert!(len <= u32::MAX, msg);
```

Reply:
`try_into` isn't a const function, so we can't use it here. And `u32::MAX as usize` is a deterministic behavior, we can ignore the `unchecked`.

Reply:
You are right, I was thinking about cases where `usize::MAX` is less than `u32::MAX`, but that is probably rare and even `ssh` itself might not run on these platforms.